File: client/web/js/lib/ErrorObject/ErrorObject.js
class ErrorObject {

    /**
     * 
     * @param {ErrorTypeEnum} ty 
     * @param {string|undefined} data 
     * @param {int} [date=new Date()] UTC in sec 
     */
    constructor(ty, data, date) {
        this.ty = ty;
        this.data = data;
        if("undefined" !== typeof(date)){
            this.date = new Date(date*1000);
        } else {
            this.date = new Date();
        }
        this.name = ErrorTypeEnumByValue[ty]
        this.msg = ErrorTypes[ty].msg;
        this.loc = ErrorTypes[ty].loc;
    }

    toString() {
        let dateStr = dateToString(this.date);
        let data = (!this.data)?"":(", data: " + this.data)
        return "Error: " + dateStr + ", " 
            + this.loc + ", " + this.ty + ", " + this.name + ", msg: " + this.msg + data;
    }
} // end of ErrorObject

ErrorObject.fromDict = function(obj) {
    return new ErrorObject(obj.code, obj.data, obj.timestamp);
}

File: server/python/error_object/error_object.py
import time

from .error_type_enum import ErrorTypeEnum
from .error_types import ErrorTypes

class ErrorObject(Exception):
    ty:ErrorTypeEnum
    data:str
    timestamp:int
    
    def __init__(self, ty:ErrorTypeEnum, data:str = None, timestamp = None):
        self.ty = ty
        self.data = data
        self.timestamp = timestamp or time.time()
    
    def toJson(self):
        return {
            "code": self.ty.value,
            "data": self.data,
            "timestamp":self.timestamp
        }
    
    def getMsg(self):
        return ErrorTypes[self.ty].msg
    
    def __str__(self):
        data = "data: {}".format(self.data) if self.data else ""
        return "Error:{} {}".format(self.getMsg(), data)